config/src/config/network_config.rs

```diff
@@ -296,3 +342,16 @@ pub struct IdentityFromStorage {
     pub key_name: String,
     pub peer_id_name: String,
 }
+
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Clone, PartialEq))]
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case", tag = "type")]
+pub enum SharedValNetAddrKeys {
+    FromConfig {
+        keys: HashMap<netaddr::KeyVersion, netaddr::Key>,
+    },
```

Comment:
for example, this is only used for testing, making this robust suggests it isn't just for testing.

Reply:
personally, i don't like it when the test model diverges from the prod model. of course, prod is currently only using one key, so maybe that's a moot point : )